bugmon.py

```diff
@@ -15,1092 +15,489 @@
-        comments.extend(bisectFixComments)
-
-      if ((bugBisectRequested or (bugBisectFixRequested and len(bisectFixComments) > 0)) and bugBisectForceCompile):
-        whiteBoardModified = True
-        wbOpts.remove('bisect-force-compile')
-        
-      if bugFailureMsg != None and bugUpdateRequested:
-        whiteBoardModified = True
-        wbOpts.remove('update')
-        comments.append(bugFailureMsg)
-
-      if whiteBoardModified:
-        wbParts = filter(lambda x: len(x) > 0, map(str.rstrip, map(str.lstrip, re.split('\[jsbugmon:[^\]]+\]', bug.whiteboard))))
-        wbParts.append("[jsbugmon:" + ",".join(wbOpts) + "]")
-
-      while True:
-        # Fetch the bug again
-        bug = self.fetchBug(bugnum)
-
-        bugModified = False
-
-        # Mark bug as WORKSFORME if confirmed to no longer reproduce
-        if closeBug:
-          bugModified = True
-          bug.status = "RESOLVED"
-          bug.resolution = "WORKSFORME"
-
-        # Mark bug as VERIFIED if we verified it successfully
-        if verifyBug:
-          bugModified = True
-          bug.status = "VERIFIED"
-
-        if whiteBoardModified:
-          # We add "ignore" to our bugmon options so we don't update the bug a second time
-          bugModified = True
-          bug.whiteboard = " ".join(wbParts)
-        
+                self._os = platform.system()
+
+        return self._os
+
+    @property
+    def arch(self):
+        """
+        Attempt to enumerate the original architecture associated with the bug
+        """
+        # ToDo: This does nothing
+        #   This needs to be modified to check if the specified bug platform matches our local platform
+        #   If not, error out
+        # if self._arch is None:
+        #     if self.bug.platform == 'ARM':
+        #         return 'ARM64'
+        #     elif self.bug.platform == 'x86':
+        #         return 'i686'
+        #     elif self.bug.platform == 'x86_64':
+        #         return 'AMD64'
+
+        return platform.machine()
+
+    @property
+    def commands(self):
+        """
+        Attempt to extract commands from whiteboard
+        """
+        commands = {}
+        if self.bug.whiteboard:
+            match = re.search(r'(?<=\[jsbugmon:).[^\]]*', self.bug.whiteboard)
+            if match is not None:
+                for command in match.group(0).split(','):
+                    if '=' in command:
+                        name, value = command.split('=')
+                        commands[name] = value
+                    else:
+                        commands[command] = None
+
+        return commands
+
+    def extract_testcase(self):
+        """
+        Attempt to extract a testcase from the bug or raise an Exception
+        """
+        attachments = list(filter(lambda a: not a.is_obsolete, self.bug.get_attachments()))
+        for attachment in sorted(attachments, key=lambda a: a.creation_time):
+            if attachment.file_name.endswith('.js'):
```

Comment:
A low-hanging fruit here would also be `.zip` files that contain e.g. a `.js` file. We use that by now for some WebAssembly crash bugs.